src/main/java/org/jenkinsci/plugins/corsfilter/AccessControlsFilter.java

```diff
@@ -147,9 +137,11 @@ public DescriptorImpl getDescriptor() {
         private String allowedHeaders;
         private String exposedHeaders;
         private String maxAge;
+        private transient List<String> allowedOriginsList;
```

Comment:
:bug: I don't think this should be marked as `transient` since we need the contents of this variable saved to disk.

Reply:
We don't need to store that in the configuration file because of is already stored in private String allowedOrigins;

Reply:
🐛 Without readResolve() logic this code is going to contain `null` after the configuration reload and hence it will fail with NPE in `isAllowed() `

Reply:
thanks @oleg-nenashev but IIUC, is not required because of it's populated inside the constructor:

              allowedOriginsList = createAllowedOriginsList(allowedOrigins);

Anyway, I'm going to add the readResolve method as per your request.